### ISSUE TITLE
Brush doesn't always update consistently

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#4779]
The brush rush the marginal histogram at the top of this [spec](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBmOgCtY3AHaQANOCgATdHkw5gkVgCcx2XAWJkhtBszace10dt2xkeWKRfpSAMwnU+LAiAEwA1vKKKgC+6hCQFPDK8DpmANoaEKAQOVAA7oyaLGYAjACsAAxxubjIjIF4ZpXVuZDISkmajErUZtk18QAefZkDUOzdfZAYgxNKsGYhFbGjA5C+jMhimmaQmlvoAJ5qqzWQeIeEyLsAjqzoSnjMOoxUJ2M5kOiDjAs4SqwxGJThAVh9IMd9CDPuhqNQjMI8NccJAkqxHu8PlALldbvdHs8nm8WmNzswxMioABhbjoxrQ6KnMFnMRHZBGdLQ-rgjBGMK7djoDkktZuCnwJ7KEZYz7sIysWCIaUyz44ymQbpIowUdCmEVYtodbhdHp-MBpSDDAC6+vBaEIF12SmU12huUZMo9Y2ZA25pLwRgesF83CMqHSBg2Yi1UzFyAlPBUKLlCqV0WiNrdUF5-P050u6sFwqgSTEod21G423aRmNkC9AwbOStox9YD98UQSQ6qSwGV9YGhkAKRSVOHKVSHKHq+FKzSH7U63V6UJllqmGy2OxR+1Zx1U2ILeIeT10ROubbWkPb602212XVgug61wP+dxKLuJ8JrwvDNtUB7uynIyh2pI5rs8CMEY8AUpia5xgmUqriq8QpoqyqofEaq7Jq7I6nqWZrPajoos6SiulhYBNh8NE1JeNRgWsAZBiGYYRre0bAXmiGSkm4zyhh6aZqh0xCrmN44SiUEwXBb6luWKKVtWSi1jsdHutCLaNgeYHDoUxQ4POazTg0c6TsRRomiu7ZDteTGypMeYzHMZpLLEg5rpu947r8z7wK+WbvuqX4Emev7waS3y-GYAJAv+Q7DCh4KwvCiLqmiGIAdhR6fvip4vMSQWnnBKI0nSkAMky+qQEBHI4GknljA58QQcmQpqE14K8YmmEGuhY62VRwW4Y8+G6pFa6Lsay5mhaxwiVhbSoA616QORlEqhpOTbQxWTFYG8xseGDWRowXH1TePXIQJqb1hm2XZuJUxSeMHXyVWilQMp+yqXWu2nNpO2jFaXqLZACKKGIbw8SkpU3klUCKkKyDqW+FLUO025XYwABe6rdPsVxKL9jQfWWl0aiTyDE6T90gNEQA) doesn't always update correctly.

Somehow one end of the brush is pretty much always null:

![Kapture 2019-03-26 at 21 53 39](https://user-images.githubusercontent.com/111269/55051295-b3a34a80-5011-11e9-9f1c-5323cae1f5c0.gif)

**Comment by kanitw** [comment]
It's quite interesting that the vertical marginal doesn't have the same problem.

**Comment by jheer** [comment]
I don't think this is a selection logic issue. It looks like a bug in the concat group generation logic. The bug arises due to a `NaN` value coming out of the `brush_x` mousemove handler: the clamp function returns `NaN` because the signal `width` is undefined. The problem is that `width` is a no-op: it is a signal whose update expression references itself!

`{"name": "width", "update": "width"}`

I think it is supposed to be referencing the top-level `width` signal in the outer scope, but that's not how signal scoping works: the top-level `width` signal is getting masked.

If I simply delete the offending signal `width` within the `concat_0_group` group, things seem to work again. Based on this analysis, I'm going to remove the assignment to @arvind.
